src/Simplify.elm

```diff
@@ -3605,6 +3719,228 @@ listIndexedMapChecks { lookupTable, fnRange, firstArg } =
                     []
 
 
+listFoldlChecks : CheckInfo -> List (Error {})
+listFoldlChecks checkInfo =
+    listFoldAnyDirectionChecks "foldl" checkInfo
+
+
+listFoldrChecks : CheckInfo -> List (Error {})
+listFoldrChecks checkInfo =
+    listFoldAnyDirectionChecks "foldr" checkInfo
+
+
+listFoldAnyDirectionChecks : String -> CheckInfo -> List (Error {})
+listFoldAnyDirectionChecks foldOperationName checkInfo =
+    -- TODO discuss adding
+    -- List.fold (&|) initial list --> initial &| (List.all identity list)
+    --         and piping versions
+    -- List.fold (&|) initial --> (List.all identity >> (&|) initial)
+    --     Are those simplifications? arguable, I'm leaning no
+    --     At the very least, they are more performant (stoppable).
+    -- List.fold f x [ a ] --> f (a) x
+    --         @jfmengels' comment: potentially quite ugly
+    --         I'd say it looks fine
+    case ( checkInfo.secondArg, checkInfo.thirdArg ) of
+        ( Just (Node initialArgumentRange _), Just (Node _ (Expression.ListExpr [])) ) ->
+            [ Rule.errorWithFix
+                { message = "The call to List." ++ foldOperationName ++ " will result in the initial accumulator"
+                , details = [ "You can replace this call by the initial accumulator." ]
+                }
+                checkInfo.fnRange
+                [ Fix.removeRange
+                    { start = initialArgumentRange.end, end = checkInfo.parentRange.end }
+                , Fix.removeRange
+                    { start = checkInfo.parentRange.start, end = initialArgumentRange.start }
+                ]
+            ]
+
+        _ ->
+            checkInfo.secondArg
+                -- when there's no initial accumulator,
+                -- we could fix to like \init -> List.sum >> (+) init
+                --     but we would then have to check for argument name clashes etc.
+                -- a point-free alternative is (>>) List.sum << (+)
+                --     but don't expect anyone to call this as simpler (except hayleigh)
+                |> Maybe.andThen
+                    (\initialArgument ->
+                        let
+                            numberBinaryOperationChecks : { identity : Int, two : String, list : String } -> List (Error {})
+                            numberBinaryOperationChecks operation =
+                                [ Rule.errorWithFix
+                                    { message = "Use List." ++ operation.list ++ " instead"
+                                    , details =
+                                        [ "Using List." ++ foldOperationName ++ " (" ++ operation.two ++ ") " ++ String.fromInt operation.identity ++ " is the same as using List." ++ operation.list ++ "." ]
+                                    }
+                                    checkInfo.fnRange
+                                    (if getUncomputedNumberValue initialArgument == Just (Basics.toFloat operation.identity) then
```

Comment:
This fix argument could definitely be extracted to a separate function

Reply:
do you mean something like
```elm
let fixWith fixes = Rule.errorWithFix ... fixes
in
```